### PR TITLE
Wrap Roberta integration test forward passes with torch.no_grad()

### LINKED ISSUE #14642: Encapsulate all forward passes of integration tests with "with torch.no_grad()"
Each model in the Transformers library has one or more corresponding integration tests, which make sure the implementation in Transformers returns the same exact logits/hidden states as the original implementation on the same input data.

e.g. the integration test for the Vision Transformer (ViT) can be found [here](https://github.com/huggingface/transformers/blob/df085d8ea8b823b85ecfe2e1968b5741983f98f3/tests/test_modeling_vit.py#L336). It verifies the shape of the logits as well as the first values on a cats image.

However, a lot of these integration tests don't leverage `with torch.no_grad()`, which make sure no gradients are computed (which isn't necessary when doing inference).

Hence, it would be great if someone could encapsulate the forward passes of the integration tests with `with torch.no_grad()`.

### PR DESCRIPTION
**Pull request by mattchurgin**
# What does this PR do?
This PR wraps forward passes in Roberta integration tests with torch.no_grad(). See issue #14642 

Fixes #14642 
[Issue link](https://github.com/huggingface/transformers/issues/14642)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [x ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

@NielsRogge @LysandreJik @sgugger

### CODE REVIEW

**Review by LysandreJik — APPROVED** [review]
Thank you for that, @mattchurgin!
